x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -155,6 +150,7 @@ public void analyzedPlan(LogicalPlan parsed, ActionListener<LogicalPlan> listene
         preAnalyze(parsed, (indices, policies) -> {
             Analyzer analyzer = new Analyzer(new AnalyzerContext(configuration, functionRegistry, indices, policies), verifier);
             var plan = analyzer.analyze(parsed);
+            plan.setAnalyzed();
```

Comment:
Now that analyzing runs in a separate sequence I've started setting this flag and checking it on the other side as an extra check.